0000-frontend-development-infrastructure-and-workflow-changes.md

```diff
@@ -0,0 +1,73 @@
+- Feature Name: frontend-development-infrastructure-and-workflow-changes
+- Start Date: 2016-07-20
+- RFC PR: 
+- Issue: 
+
+# Summary
+[summary]: #summary
+
+Currently, client side javascript in foreman is implemented with native javascript and the jquery library and managed by the Rails asset pipeline. In addition, various unrelated libraries are used for specific tasks, charting for example. This RFC proposes adding the React javascript library to the technology stack via NPM and Webpack. [(see Webpack PR)](https://github.com/theforeman/foreman/pull/3433) 
+
+# Motivation
+[motivation]: #motivation
+
+The current setup is based on software which was introduced many years ago, solved old problems that are no longer relevant and do not address the challenges that frontend developers face today. Many of the libraries we use are no longer maintained. The infrastructure is sparse and the responsibility for enforcing best practices and coding standards falls on the developer.
```

Comment:
jnotify, flot-rails, multi-select-rails (the non-rails one isn't unmaintained), gridster js is abandoned too.. there might be more - another issue is that new JS libraries aren't packaged anymore for Rails, as people tend to use npm to get them in any language